backup/README.md

```diff
@@ -0,0 +1,76 @@
+# Montagu DB backup
+
+As the database has grown (from a few 10s of MB up to GB) the dump-and-restore approach we are using is becoming painful.  Downloading from S3 takes quite a while (especially verification with duplicati) and the restore (`pg_restore`) takes a long time to rebuild the indices.
+
+## Barman
+
+This directory creates a new docker image (`vimc/montagu-barman` http://www.pgbarman.org/).
+
+Usage:
+
+```
+docker run -d --rm \
+       --name barman \
+       --network pg_nw \
+       -v barman_data:/var/lib/barman \
+       -v barman_restore:/restore \
+       docker.montagu.dide.ic.ac.uk:5000/montagu-barman:i1333
+```
+
+Important notes:
+
+The barman container must be able to reach the montagu-db container over the network.  Depending on how you are deploying this would be either by attaching to the same network (e.g., in testing) or over the network (e.g., in production)
+
+**Two** volumes are used;
+1. `barman_data` (at `/var/lib/barman`) is the one that holds the backup - this should itself be backed up
+2. `barman_restore` (at `/restore`) holds a restore location - ensure that this is empty before restoring into it
+
+Barman needs to know the hostname, port, database name, two user names and two passwords.  Defaults for these are set in [etc/barman.d.in/montagu.json](etc/barman.d.in/montagu.json) but can be overridden by passing environment variables to the container in the form `BARMAN_MONTAGU_<key>` where `<key>` is one of the entries from the json, in uppercase (this can be done either with the `-e` flag or `--env-file` argument (e.g., `BARMAN_MONTAGU_PASS_BACKUP`)
+
+Once the container is up it just runs an idle loop (which is needed to support the way that barman runs) but runs no interesting processes and does no backup
+
+Set up the initial configuration by running
+
+```
+docker exec barman setup-barman
```

Comment:
Why not run this when we start the docker container, before entering the idle loop?

Reply:
Yes, we should definitely do that